vulkaninfo/vulkaninfo.cpp

```diff
@@ -807,17 +845,24 @@ int main(int argc, char **argv) {
     if (json_output) {
         printers.push_back(std::unique_ptr<Printer>(new Printer(OutputType::json, out, selected_gpu, instance.vk_version)));
     }
+    if (vkconfig_output) {
+        vkconfig_out = std::ofstream("vulkaninfo.json");
```

Comment:
nit: Does the consumer library of this file require the ".json" extension?  If not, consider a non-descriptive "vulkaninfo.vkconfig" filename to dissuade any reverse engineers .

What is the lifetime of the file?  Who is responsible for removing the file?

Reply:
vkconfig will be responsible for the file. Right now it prints to the current directory, but a "output_path" command line argument should be added to facilitate putting `vulkaninfo.json` in a temp directory. This should be its own PR, as this one is too big already.

Reply:
Be nice not to call it .json; perhaps .bin or .vkconfig, again to emphasize its specific purpose.

the undocumented --vkconfig_output could take an argument for the path.

Reply:
I will leave it as .json for the time being to stay consistent with via.
And adding a path parameter after --vkconfig_output is not hard so I will add it to this commit so there won't be multiple versions of vulkaninfo with different behavior floating out there in the wild.